docs/quickstart.rst

```diff
@@ -41,33 +53,215 @@ Create a `virtual environment`__, and activate it. Install ``vault-cli``:
+    your terminal at some point during the tutorial, you'll need to execute this again.
+
+**We're finally ready! Yay!**
+
+Writing things in the vault
+---------------------------
+
+The vault contains secret objects, which are JSON objects (as in ``{"key": "value"}``)
+stored at specific paths. ``vault-cli`` lets us write easily write string values
+for specific keys at a specific path:
+
+.. code:: console
+
+    $ vault-cli set demo/blake2 secret_key=du9dibieNg3lei0teidal9
+    Done
+
+We can also build more complex objects. ``vault-cli`` can be passed the whole secret
+object instead of just a key/value with the flag ``--file``. Also, ``vault-cli`` can
+read its input from the standard input with the special value ``-``.
+
+``vault-cli`` expects YAML for input, and JSON is a subset of yaml, so we'll use JSON.
+
+.. code:: console
+
+    $ echo \
+        '{"Mizaru": "see no evil",' \
+        ' "Kikazaru": "hear no evil",' \
+        ' "Iwazaru": "speak no evil"}' | \
+        vault-cli set wise_monkeys --file=-
+    Done
+
+Read from the vault
+-------------------
+
+Similarily, you can read a single value from the vault:
+
+.. code:: console
+
+    $ vault-cli get demo/blake2 secret_key
+    du9dibieNg3lei0teidal9
+
+Or you can read a whole secret object at once, and receive YAML:
+
+.. code:: console
+
+    $ vault-cli get wise_monkeys
+    ---
+    Iwazaru: speak no evil
+    Kikazaru: hear no evil
+    Mizaru: see no evil
+
+You can also read multiple paths at once, recursively:
+
+.. code:: console
+
+    $ vault-cli get-all
+    ---
+    demo/blake2:
+      secret_key: du9dibieNg3lei0teidal9
+    wise_monkeys:
+      Iwazaru: speak no evil
+      Kikazaru: hear no evil
+      Mizaru: see no evil
+
+**And now, let's use vault-cli with an app!**
+
+Creating the app
+----------------
+
+We're going to try and make a basic CLI application that will hash a payload using
+Blake2_. That's not going to be the most elaborate application, but we should be able to
+showcase a standard ``vault-cli`` integration.
+
+Here's our application. We'll write it in ``quickstart_demo.py``.
 
+.. literalinclude:: quickstart_demo.py
+
+.. _Blake2: https://en.wikipedia.org/wiki/BLAKE_(hash_function)
+
+Let's look at the important parts.
+
+Settings
+~~~~~~~~
+
+.. literalinclude:: quickstart_demo.py
+   :pyobject: settings
+
+We're defining our settings, reading all the values from the environment.
+
+.. note::
+
+    Normal values are read as-is, but secret values are removed from environment as
+    they're being read. Please refer to :ref:`env-vars` for explanation.
+
+Main
+~~~~
+
+.. literalinclude:: quickstart_demo.py
+   :pyobject: main
+
+This is the core of the "app". We're using our settings without caring about their
+origin. It's important to have the application be decoupled from its configuration.
+
+Launching it as-is should result in an error because we don't define the secret key
```

Comment:
because we haven't defined?

Reply:
I had more a present tense in mind because environment variables don't necessarily stay defined. Contrary to many things that are "defined", it's possible that this env var is defined for a call, and then not defined for the following call.

I'll turn the sentence around.